src/main/java/com/spanish_inquisition/battleship/server/MessageBus.java

```diff
@@ -0,0 +1,69 @@
+package com.spanish_inquisition.battleship.server;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+public class MessageBus {
+    private Queue<Message> messageBus;
+
+    public MessageBus() {
+        messageBus = new ConcurrentLinkedQueue<>();
+    }
+
+    public void addMessage(int senderId, int recipientId, String message) {
+        Message message1 = new Message(senderId, recipientId, message);
+        messageBus.add(message1);
+    }
+
+    public Message getMessageFor(int recipientId) {
+        Message msg = null;
+        Iterator<Message> iterator = this.messageBus.iterator();
+        while (iterator.hasNext()) {
+            Message current = iterator.next();
+            if(current.getRecipient() == recipientId) {
+                msg = current;
+                iterator.remove();
+                break;
+            }
+        }
+        return msg;
+    }
+
+    public boolean haveMessageFromSender(int senderId) {
+        if(messageBus != null) {
```

Comment:
Is it necessary? If you create an object of this class you also create a ConcurrentLinkedQueue. It is empty but not null.